services/motion/builtin/builtin.go

```diff
@@ -237,7 +270,118 @@ func (ms *builtIn) MoveOnGlobe(
 	angularVelocity float64,
 	extra map[string]interface{},
 ) (bool, error) {
-	return false, ErrNotImplemented
+	operation.CancelOtherWithLabel(ctx, motionOpId)
+
+	// create a new empty framesystem which we will add our base to
+	fs := referenceframe.NewEmptyFrameSystem("")
+
+	// build maps of input enabled resources
+	_, resources, err := ms.fsService.CurrentInputs(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	// get the localizer
+	local, ok := ms.localizers[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// assert localizer as a movementSensor to get map limits
+	movementSensor, ok := local.(movementsensor.MovementSensor)
+	if !ok {
+		return false, fmt.Errorf("cannot assert local of type %T as a movementSensor", local)
+	}
+
+	currentPosition, _, err := movementSensor.Position(ctx, nil)
+	if err != nil {
+		return false, err
+	}
+	currentPose := spatialmath.GeoPointToPose(currentPosition)
+
+	// convert destination into a spatialmath pose in frame with respect to 0 latitude, 0 longitude
+	dstPose := spatialmath.GeoPointToPose(destination)
+	dstPIF := referenceframe.NewPoseInFrame(referenceframe.World, dstPose)
+
+	// convert GeoObstacles into GeometriesInFrame then into a Worldstate
+	geoms := spatialmath.GeoObstaclesToGeometries(obstacles)
+	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
+	wrldst, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	if err != nil {
+		return false, err
+	}
+	// get worldstate limits
+	limits := wrldst.BoundingBox(ctx)
```

Comment:
For GPS nav, I think we might want to allow the base to travel outside the bounds of the known obstacles. This is different from a slam map, where it shouldn't travel outside the bounds of the map. But I'm open to other opinions.

Reply:
Is there some kind of heuristic we can do for how far outside this box the robot should need to travel? Like double the size of the box and add 2x the minimum turning radius, or something?

Reply:
Yeah we'll have to go with some heuristic at the end of the day.  There are a lot of reasonable choices we could make, something like double the size of the box seems ok to me.  @biotinker can weigh in here but depending on the way TP-space RRT is implemented we may be able to safely move away from this if we only sample near the existing tree rather than sampling the entire space